### PR TITLE
get account sequence in GetTransactionCount

### PR DESCRIPTION
**Pull request by noot**
## Description

- get account sequence directly for account if querying for latest block in eth_getTransactionCount
- fixes aragon app nonce deployment issue
- add test case

test case:
```
ETHERMINT_NODE_HOST=http://localhost:8545 ETHERMINT_INTEGRATION_TEST_MODE=stable go test ./tests/... -test.v -test.run ^TestEth_GetTransactionCount
```

please test with instructions from here for the aragon app:
https://github.com/ChainSafe/ethermint/pull/299
______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/eth_api.go [hunk 1]
```diff
@@ -180,6 +180,21 @@ func (e *PublicEthAPI) GetStorageAt(address common.Address, key string, blockNum
...
+		authclient.Codec = codec.NewAppCodec(e.cliCtx.Codec)
+		accRet := authtypes.NewAccountRetriever(authclient.Codec, e.cliCtx)
+
+		_, nonce, err := accRet.GetAccountNumberSequence(from)
+		if err != nil {
+			return nil, err
+		}
+
+		n := hexutil.Uint64(nonce)
+		return &n, nil
```
Comment (fedekunze) [inline rpc/eth_api.go]: this section should be the only content of the `GetTransactionCount` function

File: rpc/eth_api.go [hunk 2]
```diff
@@ -180,6 +180,21 @@ func (e *PublicEthAPI) GetStorageAt(address common.Address, key string, blockNum
...
+	}
+
 	ctx := e.cliCtx.WithHeight(blockNum.Int64())
 	res, _, err := ctx.QueryWithData(fmt.Sprintf("custom/%s/nonce/%s", types.ModuleName, address.Hex()), nil)
```
Comment (fedekunze) [inline rpc/eth_api.go]: can we remove this query from the evm module querier too?

**Review by fedekunze — CHANGES_REQUESTED** [review]

File: rpc/eth_api.go (hunk 1 above)
Comment (noot) [reply rpc/eth_api.go]: the GetTransactiomCount function is supposed to also be query for past nonces at specific blocks as well, and this part only checks for the current nonce. 

do you know if there's another way to get the nonce at a certain height that would be better?

Comment (fedekunze) [reply rpc/eth_api.go]: You can set the height to the context using WithHeight

Comment (noot) [reply rpc/eth_api.go]: updated

File: rpc/eth_api.go (hunk 2 above)
Comment (noot) [reply rpc/eth_api.go]: yep, I removed it

**Review by fedekunze — APPROVED** [review]

**Pull request merged**